services/google_scraping_service/parser.go

```diff
@@ -0,0 +1,97 @@
+package google_scraping_service
+
+import (
+	"net/http"
+
+	"github.com/PuerkitoBio/goquery"
+)
+
+type GoogleResponseParser struct {
+	GoogleResponse *http.Response
+}
+
+type ParsingResult struct {
+	HtmlCode               string
+	LinksCount             int
+	NonAdwordsCount        int
+	NonAdwordLinks         []string
+	TopPostionAdwordsCount int
+	TopPositionAdwordLinks []string
+	TotalAdwordsCount      int
+}
+
+var ParseGoogleResponse = func(googleResp *http.Response) (ParsingResult, error) {
```

Comment:
Because I want to mock that function in the test.
There are another way to mock the function by using an interface or send the function as a parameter but I think we cannot do that in this case because this function is called by the worker instead of directly method call.

Reply:
When you explain about the mock, you refer to the use of Govcr?

Reply:
For `parser` I want to mock it for jobs testing because I want to make sure that if parse fail and reach max fails job need to update keyword status to `failed`

For `request` also the same with parser and one more thing is I am able to use VCR when I test the valid job.

You can check this out 
https://github.com/gutakk/go-google-scraper/blob/47a30a7178417b517806e5610cbf85cedca4b95b/workers/jobs/google_scraper_test.go